src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java

```diff
@@ -136,31 +138,46 @@ public static boolean shouldAnalyzeAutomatically(@Nullable VirtualFile file, @Nu
     }
 
     // file and module not null here
-    if ("java".equalsIgnoreCase(file.getFileType().getDefaultExtension()) && !isSource(file, module)) {
-      SonarLintConsole.get(module.getProject()).debug("Not automatically analysing java file outside source folder: " + file.getName());
-      return false;
-    }
-
-    return true;
-  }
-
-  public static boolean isSource(VirtualFile file, Module module) {
     ModuleRootManager moduleRootManager = ModuleRootManager.getInstance(module);
     ContentEntry[] entries = moduleRootManager.getContentEntries();
+
     for (ContentEntry e : entries) {
+      if (isExcludedOrUnderExcludedDirectory(file, e)) {
+        SonarLintConsole.get(module.getProject()).debug("Not automatically analysing excluded file: " + file.getName());
+        return false;
+      }
+
       SourceFolder[] sourceFolders = e.getSourceFolders();
       for (SourceFolder sourceFolder : sourceFolders) {
-        if (sourceFolder.getFile() == null) {
+        if (sourceFolder.getFile() == null || sourceFolder.isSynthetic()) {
           continue;
         }
+
         if (VfsUtil.isAncestor(sourceFolder.getFile(), file, false)) {
+          if (isJavaResource(sourceFolder)) {
+            SonarLintConsole.get(module.getProject()).debug("Not automatically analysing file under resources: " + file.getName());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'file' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'module' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2259)